apps/api/src/app/step-schemas/dtos/step-schema.dto.ts

```diff
@@ -1,6 +1,83 @@
 import { JSONSchema } from 'json-schema-to-ts';
+import { StepType } from '@novu/shared';
+
+import { ActionStepEnum, ChannelStepEnum } from '@novu/framework/internal';
+
+const chatProperties = {
+  body: { type: 'chatBody' },
+};
+
+const emailProperties = {
+  subject: { type: 'emailSubject' },
+  body: { type: 'emailBody' },
+};
+
+const inAppProperties = {
+  subject: { type: 'inAppSubject' },
+  body: { type: 'inAppBody' },
+  avatar: { type: 'inAppAvatar' },
+  primaryAction: { type: 'inAppPrimaryAction' },
+  secondaryAction: { type: 'inAppSecondaryAction' },
+  data: { type: 'inAppData' },
+  redirect: { type: 'inAppRedirect' },
+};
+
+const pushProperties = {
+  subject: { type: 'pushSubject' },
+  body: { type: 'pushBody' },
+};
+
+const smsProperties = {
+  body: { type: 'smsBody' },
+};
+
+const delayProperties = {
+  type: { type: 'delayType' },
+  amount: { type: 'delayAmount' },
+  unit: { type: 'delayUnit' },
+};
+
+const digestProperties = {
+  amount: { type: 'digestAmount' },
+  unit: { type: 'digestUnit' },
+  digestKey: { type: 'digestKey' },
+  lookBackWindow: { type: 'digestLookBackWindow' },
+};
+
+const customProperties = {};
+
+type Properties =
+  | typeof chatProperties
+  | typeof emailProperties
+  | typeof inAppProperties
+  | typeof pushProperties
+  | typeof smsProperties
+  | typeof delayProperties
+  | typeof digestProperties
+  | typeof customProperties;
+
+export type UiSchema = {
+  type: `${StepType}`;
+  properties?: Properties;
```

Comment:
Should properties always be supplied?
```suggestion
  properties: Properties;
```

Reply:
Furthermore, I think it would be preferable to type out the `Properties` interface separately rather than collect all the different ui schema properties into the type. That way the `Properties` interface can evolve independently from the actual UI schema definitions.

Reply:
i agree. I was not sure what approach to take here: simpler or more generic. you mean to convert the type to a generic type instead. did I get you right?